File: mobile/ios/RNCustomButtonViewManager.m
#import <Foundation/Foundation.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(RNCustomButtonViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(title, NSString)
RCT_EXPORT_VIEW_PROPERTY(disabled, BOOL)
RCT_EXPORT_VIEW_PROPERTY(onPress, RCTBubblingEventBlock)

@end

File: mobile/ios/RNCustomButtonViewManager.swift
import UIKit
import React

@objc(RNCustomButtonViewManager)
final class RNCustomButtonViewManager: RCTViewManager {
  override func view() -> UIView! {
    RNCustomButton()
  }

  override static func requiresMainQueueSetup() -> Bool {
    true
  }
}
